src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -616,7 +616,7 @@ private bool DoesTypeReferenceTypeParameter(ITypeSymbol type, ITypeParameterSymb
                 return false;
             }
 
-            if (type == typeParameter ||
+            if (type.Equals(typeParameter) ||
```

Comment:
This one is a bit subtle: it was doing direct object equality, and the type could be a wrapped type parameter with nullable annotations. Our Equals method here (on the wrapper) is now implementing the effective new behavior for symbol equals which is it ignores nullability unless you opt in.

Reply:
won't this produce a warning/error soon though?  i thought we were obsoleting ISymbol.Equals so that people would call the right version where you pass in the type of equality you want.

Should we not use htat overload and pass in the right enum value here?  at least to be more up front about what we're doing?

Reply:
Ah yeah, I guess we are, so yes, I'll update this now.